### ISSUE TITLE
Wayland fullscreen problem

### ISSUE THREAD

**Issue by kajzersoze** [dunst-project/dunst#882]
I swtiched recently from X11 to Wayland, using Sway 1.6
I'm used to setup in X11 where notifications are not shown when in fullscreen, they would be delayed and shown upon exiting the fullscreen.
In config that was done with `fullscreen = delay`, I didn't touch that config when I switched to Sway, so everything is the same and it works the same, except notifications in fullscreen, they are not delayed and are always shown.
Am I missing something special in the configuration that's needed for Wayland or that feature is not possible yet for Wayland ?

### Installation info

- Version: `Dunst - A customizable and lightweight notification-daemon 1.6.1 (2021-02-21)`
- Install type: `package`
- Distro and version: ` Gentoo`

**Comment by fwSmit** [comment]
It seems like `delay` is not working. Could you try `fullscreen = pushback` to see if that works. Otherwise, please provide the debug output with `dunst -verbosity debug`.

**Comment by kajzersoze** [comment]
No luck with `pushback`

```
INFO: Using Wayland output
INFO: New output found - id 0
DEBUG: RUN
INFO: Idle status queried: 0
INFO: Wayland: Hiding window
INFO: Adding pointer
INFO: User went idle
CRITICAL: Cannot acquire 'org.freedesktop.Notifications': Name is acquired by 'dunst' with PID '5775'.
```

**Comment by fwSmit** [comment]
Try `pkill dunst` before starting another dunst instance.

**Comment by kajzersoze** [comment]
Okay, so I did try with debug running `sleep 5; notify-send "test" "test message"` and then in mpv went fullscreen, notification was shown and this is the output, I've cut down several non-important lines like 'user went idle'

```
$ dunst -verbosity debug
INFO: Using Wayland output
INFO: New output found - id 0
DEBUG: RUN
INFO: Idle status queried: 0
INFO: Wayland: Hiding window
INFO: Adding pointer
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
DEBUG: Waking up
DEBUG: RUN
INFO: Idle status queried: 1
DEBUG: Window dimensions 196x84
DEBUG: Window position 1714x966
DEBUG: Sleeping for 9972 ms
INFO: User isn't idle anymore
INFO: User went idle
INFO: User isn't idle anymore
DEBUG: RUN
INFO: Idle status queried: 1
DEBUG: Sleeping for 0 ms
DEBUG: RUN
INFO: Idle status queried: 1
DEBUG: Queues: Closing notification for reason: time
INFO: Wayland: Hiding window
INFO: User isn't idle anymore
```

**Comment by fwSmit** [comment]
I tried it as well on v1.6.1. Turns out, the support was added after the last release. The idle behaviour was fixed as well. On second inspection `fullscreen = delay` is also working fine.
So if you want it working right now, you'll have to build from master (don't worry it's easy). Otherwise you'll have to wait a bit for the next release. It will take some time though.

**Comment by kajzersoze** [comment]
Good to hear that, well I asked in that other thread is it working on 1.6.1
Thanks!

**Comment by fwSmit** [comment]
Yeah I know, my bad

**Comment by kajzersoze** [comment]
After upgrading Sway to 1.6.1 it's still the same.

**Comment by fwSmit** [comment]
Correct, you will need to build Dunst from master to get the feature. It was added after the last release. (You also need sway version >= 1.6)

**Comment by kajzersoze** [comment]
It's not a problem for me to build from master, there's an option for that in gentoo anyway, but what I did was adding `force_xwayland = true` and it's working that way, it would do for now I guess.
If I notice some issues with using xwayland and dunst , I will switch to master :D

**Comment by kajzersoze** [comment]
Okay, so I had issues with `force_xwayland` , switched to dunst master and it works now.
Got this when I launched dunst thougn `WARNING: couldn't set idle timeout`
Which wasn't there with release version

Anyway, good job overall :+1:

**Comment by kajzersoze** [comment]
FWIW `pushback` works great, with `delay` there are issues, depending on how long is fullscreen activated it might not show notifications at all

**Comment by fwSmit** [comment]
> Okay, so I had issues with `force_xwayland` , switched to dunst master and it works now.

The xwayland version has fewer features, but other than that it should work fine.

> Got this when I launched dunst thougn `WARNING: couldn't set idle timeout`
> Which wasn't there with release version

Fixed this in #885 

> Anyway, good job overall +1

Thank you!

> FWIW `pushback` works great, with `delay` there are issues, depending on how long is fullscreen activated it might not show notifications at all

From my testing it works perfectly, even if fullscreen was activated for >30 mins.

**Comment by kajzersoze** [comment]
Might be my mistake regarding `delay`, it probably is, maybe I mixed few things up when I was still configuring some things, but it was definitely happening, but was it with release version, was it with xwayland I can't remember, I thought it was because `delay` so I switched to `pushback` and I left it like that.
I'm lazy to test that out, because it really doesn't matter to me now, but I'll keep it in mind and I might try it eventually.
